src/templates/index.jsx

```diff
@@ -453,13 +461,23 @@ const Layout = React.createClass({
         this.noteListVisibilityUpdater = updater;
     },
 
+    storeDetailViewRef(ref) {
+        this.detailView = ref;
+    },
+
     render() {
+        const { previouslySelectedNote, selectedNote } = this.state;
+
         const currentNote = find(
             this.props.notes,
             matchesProperty('id', this.props.selectedNoteId)
         );
         const filteredNotes = this.filterController.getFilteredNotes(this.props.notes);
 
+        if (this.detailView && selectedNote !== previouslySelectedNote) {
```

Comment:
parens here would be helpful. it's not obvious to me that JS will do what we want

```js
(this.detailView && selectedNote) !== previouslySelectedNote

// vs.

this.detailView && ( selectedNote !== previouslySelectedNote )
```